packages/tailwindcss/src/index.ts

```diff
@@ -61,15 +61,49 @@ async function parseCss(
 
     // Collect paths from `@plugin` at-rules
     if (node.selector === '@plugin' || node.selector.startsWith('@plugin ')) {
-      if (node.nodes.length > 0) {
-        throw new Error('`@plugin` cannot have a body.')
-      }
-
       if (parent !== null) {
         throw new Error('`@plugin` cannot be nested.')
       }
 
-      pluginPaths.push(node.selector.slice(9, -1))
+      let options: CssPluginOptions = {}
+
+      for (let decl of node.nodes ?? []) {
+        if (decl.kind !== 'declaration') {
+          throw new Error('`@plugin` can only contain declarations.')
```

Comment:
Some ideas:

- Should this log what we found instead? (`@plugin` can only contain a flat list of declarations, instead found rule `bar { ... }`).
- Maybe if we detect a rule and it looks like they want to pass an object, we can explain that this is not supported and that they can do this in a `.js` file only for now or so?

Reply:
yeah good idea